src/index.ts

```diff
@@ -194,28 +194,25 @@ export function getBuildThroughStream(opts: TransposeOptions): NodeJS.ReadWriteS
 
 	// Regex to match against the type of command, e.g. FROM, RUN, COPY
 	const stepCommandRegex = /^\s?(\w+)(:?\s)/i
-	const getStepCommand = RegExp.prototype.exec.bind(stepCommandRegex)
+	// Use type-coercion here to suppress TS `may be undefined` warnings, as we know
+	// that function is only called with a value that will produce a non-undefined value
+	const getStepCommand = (str: string) => stepCommandRegex.exec(str)![1].toUpperCase()
 
 	// Regex to remove extra flags which this module adds in
-	const replaceRegex = new RegExp(`${opts.containerQemuPath} -execve /bin/sh -c `, 'i')
+	const replaceRegexString = _.escapeRegExp(`${opts.containerQemuPath} -execve /bin/sh -c `)
+	const replaceRegex = new RegExp(replaceRegexString, 'i')
 	const replaceQemuLine = (data: string): string => {
 		return data.replace(replaceRegex, '')
 	}
 
 	return es.pipe(
-		// Ensure we get one line at a time
-		es.split(),
-		es.through(function(data: string) {
-
-			if (isStepLine(data)) {
-				if (getStepCommand(stripStepPrefix(data))[1].toUpperCase() === 'RUN') {
-					this.emit('data', replaceQemuLine(data) + EOL)
-				} else {
-					this.emit('data', data + EOL)
-				}
-			} else {
-				this.emit('data', data + EOL)
+		es.mapSync(function(data: string) {
+
+			if (isStepLine(data) && getStepCommand(stripStepPrefix(data)) === 'RUN') {
+				data = replaceQemuLine(data)
 			}
-		})
+			this.emit('data', data)
+		}),
+		es.join()
```

Comment:
I was basing it on [this comment](https://github.com/resin-io-modules/docker-qemu-transpose/pull/4#discussion_r114006032) but you're right, it needs the join char to be explicitly provided.